core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -118,4 +119,62 @@ class ReplicaManagerTest {
 
     TestUtils.verifyNonDaemonThreadsStatus(this.getClass.getName)
   }
+
+  @Test
+  def testClearPurgatoryOnBecomingFollower() {
+    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect)
+    props.put("log.dir", TestUtils.tempRelativeDir("data").getAbsolutePath)
+    val config = KafkaConfig.fromProps(props)
+    val zkClient = EasyMock.createMock(classOf[ZkClient])
+    val zkUtils = ZkUtils(zkClient, false)
```

Comment:
This is another instance where a named arguments can be used (same as the example mentioned by @jjkoshy).